pkg/apis/monitoring/v1/prometheus_types.go

```diff
@@ -2129,6 +2129,10 @@ type ScrapeClass struct {
 	// +optional
 	TLSConfig *TLSConfig `json:"tlsConfig,omitempty"`
 
+	// Authorization section for the ScrapeClass.
+	// +optional
+	Authorization *Authorization `json:"authorization,omitempty"`
```

Comment:
we would need to call Authorization.Validate() in getScrapeClassConfig().
_looking at the existing code, we forgot to do it for TLSConfig._